integration-test/build.gradle

```diff
@@ -19,6 +21,21 @@ dependencies {
     testRuntimeOnly "org.junit.vintage:junit-vintage-engine:5.11.4"
 }
 
+// Prevent newer Gradle from switching to JRE version of Guava for building/running integration tests
+// We want our tests to run on the lowest-common denominator
+def gradleVersionTargetJVM = GradleVersion.version("7.0")
+if (GradleVersion.current() > gradleVersionTargetJVM) {
+    dependencies.constraints {
+        implementation("com.google.guava:guava") {
```

Comment:
According to the commit comment, this should be `testImplementation()` too?

Reply:
Since this module is a test module and nothing will depend on it, I deliberately chose `implementation` so the Android version would be used even in the regular compile phase of this module. But the comment is still accurate because it is a test module.

I can see how this might look like a mistake though. It would hurt to change it to `testImplementation` as I don't think there is any non-test code to compile. Alternatively I could update the comment to clarify this.

Reply:
You are correct, the _commit comment_ was wrong. I have improved the code comments and fixed the commit comment.